src/processor_arm.cpp

```diff
@@ -1187,6 +1199,18 @@ static NOINLINE std::pair<uint32_t,FeatureList<feature_sz>> _get_host_cpu()
     // Here we assume that only the lower 32bit are used on aarch64
     // Change the cast here when that's not the case anymore (and when there's features in the
     // high bits that we want to detect).
+#ifdef _CPU_AARCH64_ && _OS_DARWIN_
+    CPUID info = {
+            uint8_t(0x61),
+            uint8_t(0),
+            uint16_t(0x21)
```

Comment:
I don't think you should try to reuse this code and should instead replace the whole function (`_get_host_cpu`).

But if you do, you should use the M1 info, i.e. `0x23`.